UnitedIncome/components
# Standardizes Z-Index Values

### Description
> Please provide a brief description of your changes below.

Makes a change which standardizes z-indexes on the following scale:

```sass
// z-index scale in ascending order.
$z-index-auto: auto;
$z-index-below: -1;
$z-index-default: 1;
$z-index-popup: 4000;
$z-index-dialog: 5000;
$z-index-dropdown: 6000;
$z-index-overlay: 7000;
$z-index-menu: 8000;
$z-index-modal: 9000;
$z-index-toast: 10000;
```

I've started to these these in client and so far so good. These get exported via Webpack so you can import either the Sass variables, JavaScript variables, or you can just use the provided helper classes. I figured all three was appropriate as we do something similar for colors.

I've also created a story which stacks these items, that way we can catch regressions early if something ends up changing with these values (though they shouldn't change). I will need to update the main README file if everyone is ok with this. 

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* Load up the `utilities/Styles/zindex` story to see this in action.


### Comments
> Any additional comments that will aid with the review process.

There will be a subsequent change that will come soon after to fully update the dotcom and client sites to use these variables. 

There's some occurrences where there's a single digit increment on the z-index. I think using Sass to do this as an exception is a fine pattern so long as it's by `1` and not `1000` or something ridiculous. 

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-*)
